src/melee.cpp

```diff
@@ -1406,6 +1410,13 @@ bool player::block_hit(Creature *source, body_part &bp_hit, damage_instance &dam
         melee_attack( *source, false, tec );
     }
 
+    // give stamina penalty for blocking, heavier weapons and arm encumbrance
+    // are more tiring, while higher block bonus means less effort expended
+    int encumbrance_cost = roll_remainder( ( encumb( bp_arm_l ) + encumb( bp_arm_r ) ) / 5.0f );
+    int weight_cost = weapon.weight() / ( 20 * std::max( 1, str_cur ) );
```

Comment:
Effects of fixing this are that stamina drain for shields went up massively. Now changing it so that the block bonus is considered in full instead of being halved.